File: SEP3_Database/src/main/java/model/domain/unit/user/Friend.java
package model.domain.unit.user;

public class Friend extends BasicInformation {

    public Friend(Account account) {
        super(account.getId(), account.getUserName());
        setBirthday(account.getBirthday());
    }

    public Friend(String id, String userName, Birthday birthday) {
        super(id, userName);
        setBirthday(birthday);
    }

    public void update(Friend friend) {
        update(friend.getUserName(),friend.getBirthday());
    }

    public Friend copy() {
        return new Friend(getId(),getUserName(),getBirthday());
    }
}

File: SEP3_Client1/Model/Unit/User/Friend.cs
using System;

namespace SEP3_Client.Model.Unit.User
{
    public class Friend : BasicInformation
    {
        public Friend(Account account) : base(account.GetId(),account.GetUserName())
        {
            SetBirthday(account.GetBirthday());
        }

        public Friend(string id, string userName, Birthday birthday) : base(id, userName)
        {
            SetBirthday(birthday);
        }
        
        public Friend() {}

        public void Update(Friend friend) 
        {
            Update(friend.GetUserName(),friend.GetBirthday(),friend.Gender,friend.Hometown,friend.PhoneNumber);
        }
        
        public Friend Copy() 
        {
            return new Friend(GetId(),GetUserName(),GetBirthday());
        }
    }
}